allauth_2fa/forms.py

```diff
@@ -59,27 +60,29 @@ def save(self):
         return self.device
 
 
-class TOTPDeviceRemoveForm(forms.Form):
+class TOTPDeviceRemoveForm(OTPAuthenticationFormMixin, forms.Form):
     # User must input a valid token so 2FA can be removed
-    token = forms.CharField(
+    otp_token = forms.CharField(
         label=_("Token"),
     )
 
     def __init__(self, user, **kwargs):
         super().__init__(**kwargs)
-        self.user = user
-        self.fields["token"].widget.attrs.update(DEFAULT_TOKEN_WIDGET_ATTRS)
 
-    def clean_token(self):
-        # Ensure that the user has provided a valid token
-        token = self.cleaned_data.get("token")
+        if self.fields.get("token") or (
+            kwargs.get("data") and "token" in kwargs.get("data")
+        ):
```

Comment:
We already called `form.Form.__init__` so `self.data` is a thing, so
```suggestion
        if "token" in self.fields or "token" in self.data:
```
should suffice?